packages/modules/payment/src/services/payment-module.ts

```diff
@@ -294,33 +294,42 @@ export default class PaymentModuleService
     input: CreatePaymentSessionDTO,
     @MedusaContext() sharedContext?: Context
   ): Promise<PaymentSessionDTO> {
-    let paymentSession: PaymentSession
+    let paymentSession: PaymentSession | undefined
 
     try {
+      paymentSession = await this.createPaymentSession_(
+        paymentCollectionId,
+        input,
+        sharedContext
+      )
+
       const providerSessionSession =
         await this.paymentProviderService_.createSession(input.provider_id, {
-          context: input.context ?? {},
+          context: { ...input.context, resource_id: paymentSession.id },
```

Comment:
thought: Should the `resource_id` be overwritable? In case you want to associate the cart instead of the payment session. The Medusa payment session will always have a direct association with the third-party session through its `payment_session.data`.

Reply:
I think we can use a more specific field name to associate the Medusa payment session with the provider session, so as to avoid unintentional overwriting.

Reply:
`session_id`?

Reply:
> `session_id`?

it’s ok

Reply:
I removed resource_id and replaced it with session_id. If other resource identifiers like order_id or cart_id are needed, I think they can be added based on future requirements, as currently, only session_id is the most useful.